### ISSUE TITLE
ITMS-90683: Missing Purpose String in Info.plist 

### ISSUE THREAD

**Issue by tamsel12** [dpa99c/cordova-diagnostic-plugin#369]
Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSBluetoothAlwaysUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data are required to include a purpose string. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. 

Getting this error while uploading IOS app to appstore. anybody has solution for this ? 

Thanks in Advance !

**Comment by samih-dev** [comment]
Hi,

facing the same issue which blocks uploading new versions to App Store.

Another StackOverflow thread on the same issue: 
https://stackoverflow.com/questions/58009162/ios-app-store-nsbluetoothalwaysusagedescription-rejection

unfortunately, I have no experience dealing with Cordova plugins code, so waiting for someone help,

The solution may be similar to how **cordova-plugin-googlemaps** expose variables to set when installing:
https://github.com/mapsplugin/cordova-plugin-googlemaps#install-optional-variables

Thanks,

**Comment by kamilbrk** [comment]
You can temporarily fix the issue by yourself by adding the following to your `config.xml` file:

```xml
        <config-file parent="NSBluetoothAlwaysUsageDescription" platform="ios" target="*-Info.plist">
            <string>Bluetooth permission is required because of X, Y and Z</string>
        </config-file>
```

On supported Cordova versions, that is. Look up `config-file`, `edit-config` etc for more information.

This will add an additional line into your `plist` file upon build, a line that's currently not provided out of the box by the plugin.

If you do not use Bluetooth module of this plugin at all, you can also remove that module altogether, see readme file for instructions. If you've never selected specific modules to install, simply add

```xml
    <preference name="cordova.plugins.diagnostic.modules" value="LOCATION WIFI CAMERA NOTIFICATIONS MICROPHONE CONTACTS CALENDAR REMINDERS MOTION NFC EXTERNAL_STORAGE" />
```

to your `config.xml` file. This will ignore the Bluetooth module whilst still adding everything else, so the change will be minimal from the default installation behaviour. Again, see documentation. When that's done, remove and add the plugin again. In case you are using any build process/scripts, make sure to clear `node_modules` as well. I've had to `rm -rf node_modules platforms plugins` and then `npm i` which triggers `cordova prepare` via `postinstall` script. Your setup may vary, but essentially after clearing things up you should end up with the following:

<img width="595" alt="Screenshot 2019-09-27 at 09 48 23" src="https://user-images.githubusercontent.com/591349/65769574-c7984800-e12b-11e9-8300-84b93db34a97.png">

**Comment by samih-dev** [comment]
thanks @kamilbrk 

I was able to submit to the App Store without issues passing Apple Initial tests. waiting for final approval from Apple.

**Comment by dpa99c** [comment]
This has been fixed in v5.0.1

**Comment by stegithub** [comment]
i've the same issue with v5.0.1.
have i to implement @kamilbrk workaround?
